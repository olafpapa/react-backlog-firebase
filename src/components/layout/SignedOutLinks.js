import * as React from 'react';
import { NavLink } from 'react-router-dom'

export const SignedOutLinks = () => {
  return (
    <ul className="right">
      <li><NavLink to='/signup'>Sign Up</NavLink></li>
      <li><NavLink to='/signin'>Login</NavLink></li>
    </ul>
  );
};

export default SignedOutLinks
