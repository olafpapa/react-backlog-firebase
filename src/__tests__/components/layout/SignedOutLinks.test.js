import React from 'react'
import { BrowserRouter, NavLink } from 'react-router-dom'
import  { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import SignedOutLinks from '../../../components/layout/SignedOutLinks'

describe('<SignedOutLinks />', () => {

  it('子コンポーネントが存在すること', () => {
    // == 準備 ==
    /** Appコンポーネントをshallowレンダリング */
    const wrapper = shallow(<SignedOutLinks/>);

    // == 検証 ==
    /** コンポーネントの数を取得し、2であればOK */
    expect(wrapper.find(NavLink).length).toBe(2);
  });

  test('<SignedOutLinks />のスナップショット', () => {
    const tree = renderer
      .create(<BrowserRouter><SignedOutLinks /></BrowserRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
