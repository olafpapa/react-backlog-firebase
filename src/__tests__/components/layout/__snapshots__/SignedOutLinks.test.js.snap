// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignedOutLinks /> <SignedOutLinks />のスナップショット 1`] = `
<ul
  className="right"
>
  <li>
    <a
      aria-current={null}
      href="/signup"
      onClick={[Function]}
    >
      Sign Up
    </a>
  </li>
  <li>
    <a
      aria-current={null}
      href="/signin"
      onClick={[Function]}
    >
      Login
    </a>
  </li>
</ul>
`;
